crates/oxc_minifier/src/ast_passes/peephole_minimize_conditions.rs

```diff
@@ -275,6 +275,9 @@ impl<'a> PeepholeMinimizeConditions {
                 Expression::BooleanLiteral(consequent_lit),
             ) = (&expr.left, &expr.right)
             {
+                if !is_in_boolean_context(ctx) {
+                    return None;
+                }
```

Comment:
there was a bug here before.

it's not safe to minify `a && true` to `a` in cases where you're not inside a node that will be coerced to a boolean

e.g.
```ts
var a = '123'
a && true; // evaluates to true
a; // evaulates to '123'
```